Python/Product/TestAdapter.Executor/UnitTest/UnitTestExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.PythonTools.TestAdapter.UnitTest {
+    public static class UnitTestExtensions {
+
+        public static TestCase ToVsTestCase(this UnitTestTestCase test, bool isWorkspace, string projectHome) {
+            var normalizedPath = test.Source.ToLowerInvariant();
```

Comment:
after checking out the javascript test adapter I sort of liked how they converted discovered tests to vsTests using an extension class.. https://github.com/mmanela/chutzpah

I'm now thinking of updating the pytest discover flow to do something similar